Expensify/App
# Added Permission check before setting Preferred Language

### Details


### Fixed Issues

$ https://github.com/Expensify/App/issues/5286

### Tests | QA Steps

1. Launch the app
2. Log in with an account which is Spanish (expensifail domain account)
3. Log out from account
4. Log in with an account which is English (Expensifail or gmail account)
5. Go to Settings
6. Check the language of the app


### Tested On


### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
